engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -28,8 +28,9 @@
 @ExportLibrary(TypesLibrary.class)
 @ExportLibrary(InteropLibrary.class)
 public final class Type implements EnsoObject {
+
   private final String name;
-  private @CompilerDirectives.CompilationFinal ModuleScope definitionScope;
+  private @CompilerDirectives.CompilationFinal ModuleScope.Builder definitionScope;
```

Comment:
DOes it make sense to declare fields of data as `@CompilationFinal`?

Reply:
It might make sense. Instances of `Type` are directly referenced from various Enso `Node` - e.g. they are part of the AST. Remember we have the [EXCLUSIVE](https://github.com/enso-org/enso/blob/202f7e1a4950fe77b4c7bb746e83c98e7d256750/engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java#L70) sharing policy - there is no sharing of ASTs.